cantools/database/can/c_source.py

```diff
@@ -1503,6 +1757,10 @@ def _generate_helpers(kinds):
                                             UNPACK_HELPER_RIGHT_SHIFT_FMT)
     helpers = pack_helpers + unpack_helpers
 
+    helpers.append("#define SNPRINTPF_BUFFER_LEN 256")
+    helpers.append("// Buffer for formatting JSON messages.")
+    helpers.append("char snprintf_buffer[SNPRINTPF_BUFFER_LEN];\n\n")
```

Comment:
Does this end up on the stack of the decoding function? Or is it a global heap variable? If it's the latter, I'd prefer it to be the former instead to avoid multithreading issues.

Reply:
It's currently a global shared buffer which would be multithreading-unfriendly. I've moved it to be a per-function instance.